src/main/java/dev/hephaestus/atmosfera/world/context/Sphere.java

```diff
@@ -126,6 +132,39 @@ public void update(ClientPlayerEntity player) {
                 mut.move(Direction.DOWN);
             }
 
+            BossBarHud bossBarHud = MinecraftClient.getInstance().inGameHud.getBossBarHud();
+            Map<UUID, ClientBossBar> bossBars = ((BossBarHudAccessor) bossBarHud).getBossBars();
+
+            this.isInRaid = false;
+            this.isDefeatedInRaid = false;
+            this.isVictoriousInRaid = false;
+            this.isInWitherFight = false;
+            this.isInEnderDragonFight = false;
+            if (!bossBars.isEmpty()) {
+                Iterator var1 = bossBars.values().iterator();
+
+                while(var1.hasNext()) {
+                    BossBar bossBar = (BossBar)var1.next();
+                    MutableText bossBarText = bossBar.getName().copy();
+                    if(bossBarText.toString().contains("'event.minecraft.raid'")) {
```

Comment:
This logic will end up moving to the actual condition, but calling `toString` isn't necessary here either. I'd do something like
```java
String value = bossBarText instanceof TranslatableText translatable ? translatable.getKey() : bossBarText.asString();
```